plugins/degu.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Plugins for using Degu elements.
+ */
+
+import {NunjucksTemplateEngine, Pod} from '@amagaki/amagaki';
+
+import nunjucks from 'nunjucks';
+
+export class DeguPlugin {
+  static register(pod: Pod) {
+    const nunjucksEngine = pod.engines.getEngineByExtension(
+      '.njk'
+    ) as NunjucksTemplateEngine;
+    nunjucksEngine.env.addGlobal('degu', {
+      asset: DeguPlugin.asset,
+    });
+  }
+
+  static asset(options: {
+    url: string;
+    alt: string;
+    width?: number;
```

Comment:
This not a specific action item for this PR but I think it would be good to figure out / think about ways to populate the width / height from backend / asset pipeline and automate this.  In one of the projects, we had created a map of urls to asset meta info so the lookup be automated.

Reply:
Yeah; I think that should be external to this macro/template function but it's something that we should definitely provide a generic approach for. I think it might be a custom YAML tag with a hook that runs prior to building that ensures all of the referenced files are uploaded to the images service.